src/coreclr/jit/compiler.h

```diff
@@ -8745,7 +8767,11 @@ XXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXX
     var_types getSIMDVectorType()
     {
 #if defined(TARGET_XARCH)
-        if (getSIMDSupportLevel() == SIMD_AVX2_Supported)
+        if (getSIMDSupportLevel() == SIMD_AVX512F_Supported)
+        {
+            return TYP_SIMD64;
+        }
+        else if (getSIMDSupportLevel() == SIMD_AVX2_Supported)
```

Comment:
```suggestion
        // TODO-XArch-AVX512 : Return TYP_SIMD64 once Vector<T> supports AVX512.
        if (getSIMDSupportLevel() == SIMD_AVX2_Supported)
```

Reply:
Actually looks like the reason this isn't causing failures is because it is also effectively dead code like the case I mentioned above.

`getSIMDVectorRegisterByteLength` is the only one that is meaningfully used anymore.

Reply:
The one place it is used could be replaced with an `assert(getSIMDVectorRegisterByteLength() == 16)`

Reply:
Interesting. I thought the check for `Vector<T>` is https://github.com/dotnet/runtime/blob/6ede02b93ce17925158486729be53cad34010c4f/src/coreclr/jit/compiler.h#L8784-L8786. Will update this too just to be safe

Reply:
They originally all were used in coordination, but we've effectively removed all the usages of the other two. So this one is basically dead code now. Previously it was used to determine what instruction sets were valid for use as part of handling Vector<T> and there were various asserts that the byte length and feature level were equivalent.

Reply:
Okay. Though think it needs to be `if (getSIMDSupportLevel() >= SIMD_AVX2_Supported)` since it returns `SIMD_AVX512F_Supported `too now